Propagate all user methods in REST Data with Panache

Sgitario opened the pull request:
Relates to https://github.com/quarkusio/quarkus/pull/35129#discussion_r1280359385 and https://github.com/quarkusio/quarkus/pull/35129

manovotn commented on extensions/panache/rest-data-panache/deployment/src/main/java/io/quarkus/rest/data/panache/deployment/methods/UserMethodsImplementor.java:
```diff
@@ -14,26 +11,17 @@
             for (var methodInfo : resourceMetadata.getResourceInterface().methods()) {
-                // we only need to propagate the user methods annotated with `@Transactional`
-                if (methodInfo.hasAnnotation(TRANSACTIONAL)) {
+                if (methodInfo.isDefault()) {
```
IIUIC, with this approach you are copying over all default methods and their annotations.
This definitely solves the bindings, I have no issue with that. 

However, it also carries over different annotations that would otherwise be ignored - such as `@Produces`. I am not sure that's something of a concern here, most likely not :)
Last but not least, you are very likely copying methods (most of them?) that you wouldn't even need to copy as they have either no annotation or no relevant annoation. A simple check for method having *any* annotation while being default might improve that?

Sgitario replied:
> A simple check for method having any annotation while being default might improve that?

this makes total sense. will update it.

Sgitario committed:
Propagate all user methods in REST Data with Panache

Relates to https://github.com/quarkusio/quarkus/pull/35129#discussion_r1280359385 and https://github.com/quarkusio/quarkus/pull/35129

Sgitario commented:
PR updated

mkouba reviewed (approved):
> However, it also carries over different annotations that would otherwise be ignored - such as @Produces. I am not sure that's something of a concern here, most likely not :)

I think that we don't expect annotations like `@Produces` to be used here.

manovotn commented:
Thanks @Sgitario
